plugins/flytekit-kf-pytorch/tests/test_elastic_task.py

```diff
@@ -187,3 +188,38 @@ def wf(recoverable: bool):
     else:
         with pytest.raises(RuntimeError):
             wf(recoverable=recoverable)
+
+
+def test_run_policy() -> None:
+    """Test that run policy is propagated to custom spec."""
+    from flytekitplugins.kfpytorch.task import PytorchElasticFunctionTask
+
+    run_policy = RunPolicy(
+        clean_pod_policy=CleanPodPolicy.ALL,
+        ttl_seconds_after_finished=10 * 60,
+        active_deadline_seconds=36000,
+        backoff_limit=None,
+    )
+
+    pytorch_tasks = []
+    original_init = PytorchElasticFunctionTask.__init__
+
+    def mock_init(self, *args, **kwargs) -> None:
```

Comment:
Thanks for adding a test that ensures that the run policy is correctly set by `.get_custom()`.

I find the mocking logic unnecessarily complex. [This](https://github.com/flyteorg/flytekit/blob/2719b349cd6ac9b18cad858ff2fc6d6f79986b1e/plugins/flytekit-kf-pytorch/tests/test_pytorch_task.py#L143) test for the `PytorchFunctionTask` shows how you can call `.get_custom` on a task without mocking the constructor. Can you please simplify it here?

Reply:
Addressed